src/ext/inline_autocomplete_test.js

```diff
@@ -0,0 +1,286 @@
+
+if (typeof process !== "undefined") {
+    require("amd-loader");
+    require("../test/mockdom");
+}
+
+"use strict";
+
+var ace = require("../ace");
+var Editor = require("../editor").Editor;
+var EditSession = require("../edit_session").EditSession;
+var InlineAutocomplete = require("./inline_autocomplete").InlineAutocomplete;
+var assert = require("../test/assertions");
+var type = require("../test/user").type;
+var VirtualRenderer = require("../virtual_renderer").VirtualRenderer;
+
+var editor;
+var autocomplete;
+
+var getAllLines = function() {
+    var text = Array.from(editor.renderer.$textLayer.element.childNodes).map(function (node) {
+        return node.textContent;
+    }).join("\n");
+    if (editor.renderer.$ghostTextWidget) {
+        return text + "\n" + editor.renderer.$ghostTextWidget.text;
+    }
+    return text;
+};
+
+var typeAndChange = function(...args) {
+    assert.equal(autocomplete.changeTimer.isPending(), null);
+    type(args);
+    assert.ok(autocomplete.changeTimer.isPending);
+    autocomplete.changeTimer.call();
+};
+
+var completions = [
+    {
+        value: "foo"
+    },
+    {
+        value: "foobar"
+    },
+    {
+        value: "function"
+    },
+    {
+        value: "fundraiser"
+    },
+    {
+        snippet: "function foo() {\n    console.log('test');\n}",
+        caption: "func"
+    },
+    {
+        value: "foobar2"
+    }
+];
+
+var mockCompleter = {
+    getCompletions: function(_1, _2, _3, _4, callback) {
+        callback(null, completions);
+    }
+};
+
+module.exports = {
+    setUp: function(done) {
+        var el = document.createElement("div");
+        el.style.left = "20px";
+        el.style.top = "30px";
+        el.style.width = "500px";
+        el.style.height = "500px";
+        document.body.appendChild(el);
+        var renderer = new VirtualRenderer(el);
+        var session = new EditSession("");
+        editor = new Editor(renderer, session);
```

Comment:
Should we test more than one editor too?

Reply:
I adjusted this class to be editor scoped.